kaspad/src/daemon.rs

```diff
@@ -0,0 +1,357 @@
+use kaspa_core::{kaspad_env::version, task::tick::TickService};
+use kaspa_grpc_server::service::GrpcService;
+use kaspa_rpc_service::service::RpcCoreService;
+use kaspa_utils::networking::ContextualNetAddress;
+
+use kaspa_addressmanager::AddressManager;
+use kaspa_consensus::pipeline::monitor::ConsensusMonitor;
+use kaspa_consensus::{consensus::factory::Factory as ConsensusFactory, pipeline::ProcessingCounters};
+use kaspa_consensusmanager::ConsensusManager;
+use kaspa_core::task::runtime::AsyncRuntime;
+use kaspa_index_processor::service::IndexService;
+use kaspa_mining::manager::{MiningManager, MiningManagerProxy};
+use kaspa_p2p_flows::{flow_context::FlowContext, service::P2pService};
+
+use kaspa_perf_monitor::builder::Builder as PerfMonitorBuilder;
+use kaspa_utxoindex::{api::UtxoIndexProxy, UtxoIndex};
+use kaspa_wrpc_server::service::{Options as WrpcServerOptions, ServerCounters as WrpcServerCounters, WrpcEncoding, WrpcService};
+
+use crate::args::Args;
+
+const DEFAULT_DATA_DIR: &str = "datadir";
+const CONSENSUS_DB: &str = "consensus";
+const UTXOINDEX_DB: &str = "utxoindex";
+const META_DB: &str = "meta";
+const DEFAULT_LOG_DIR: &str = "logs";
+
+fn get_home_dir() -> PathBuf {
+    #[cfg(target_os = "windows")]
+    return dirs::data_local_dir().unwrap();
+    #[cfg(not(target_os = "windows"))]
+    return dirs::home_dir().unwrap();
+}
+
+fn get_app_dir() -> PathBuf {
+    #[cfg(target_os = "windows")]
+    return get_home_dir().join("rusty-kaspa");
+    #[cfg(not(target_os = "windows"))]
+    return get_home_dir().join(".rusty-kaspa");
+}
+
+fn validate_config_and_args(_config: &Arc<Config>, args: &Args) -> ConfigResult<()> {
+    if !args.connect_peers.is_empty() && !args.add_peers.is_empty() {
+        return Err(ConfigError::MixedConnectAndAddPeers);
+    }
+    if args.logdir.is_some() && args.no_log_files {
+        return Err(ConfigError::MixedLogDirAndNoLogFiles);
+    }
+    Ok(())
+}
+
+fn get_user_approval_or_exit(message: &str, approve: bool) {
+    if approve {
+        return;
+    }
+    println!("{}", message);
+    let mut input = String::new();
+    match std::io::stdin().read_line(&mut input) {
+        Ok(_) => {
+            let lower = input.to_lowercase();
+            let answer = lower.as_str().strip_suffix("\r\n").or(lower.as_str().strip_suffix('\n')).unwrap_or(lower.as_str());
+            if answer == "y" || answer == "yes" {
+                // return
+            } else {
+                println!("Operation was rejected ({}), exiting..", answer);
+                exit(1);
+            }
+        }
+        Err(error) => {
+            println!("Error reading from console: {error}, exiting..");
+            exit(1);
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct Runtime {
+    log_dir: Option<String>,
+}
+
+impl Runtime {
+    pub fn from_args(args: &Args) -> Self {
+        // Configure the panic behavior
+        kaspa_core::panic::configure_panic();
+
+        let network = args.network();
```

Comment:
Seems like lines 94-107 can be removed from here. They are repeated at `create_core_with_runtime`

Reply:
* network should be an argument to Runtime, and then you can use `network.name()` instead of `config.network_name()`